ampersand-filtered-subcollection.js

```diff
@@ -20,7 +20,7 @@ var slice = Array.prototype.slice;
 
 function FilteredCollection(collection, spec) {
     this.collection = collection;
-    this.indexes = collection.indexes;
+    this.indexes = collection.indexes || [];
```

Comment:
IMO it's flawed that it doesn't default to an empty array on initialize in collection tbh.